package.json

```diff
@@ -128,10 +126,12 @@
     "vanilla-framework": "^3.15.1 || ^4.0.0"
   },
   "scripts": {
-    "build": "rm -rf dist && yarn build-local; yarn build-declaration",
-    "build-local": "NODE_ENV=production babel src --out-dir dist --copy-files --no-copy-ignored --extensions '.js,.jsx,.ts,.tsx,.snap' --ignore '**/*.test.ts','**/*.test.tsx','**/*.stories.tsx','**/__snapshots__','src/setupTests.js','src/testing'",
-    "build-declaration": "tsc --project tsconfig.build.json && tsc-alias -p tsconfig.build.json",
-    "build-watch": "yarn run build-local --watch",
+    "build": "rm -rf dist && yarn build-cjs && yarn build-declaration && yarn build-esm && yarn build-declaration-esm",
+    "build-cjs": "NODE_ENV=production BABEL_ENV=cjs babel src --out-dir dist --copy-files --extensions '.js,.jsx,.ts,.tsx'",
+    "build-declaration": "tsc --project tsconfig.json && tsc-alias -p tsconfig.json",
+    "build-esm": "NODE_ENV=production BABEL_ENV=esm babel src --out-dir dist/esm --copy-files --extensions '.js,.jsx,.ts,.tsx'",
+    "build-declaration-esm": "tsc --project tsconfig.json --outDir dist/esm && tsc-alias -p tsconfig.json --outDir dist/esm",
```

Comment:
Having generated types only in the root folder `/dist` doesn't work for imports from `/dist/esm` even if the package.json has this set: `"types": "dist/types/index.d.ts"` because in the case of ESM import the import path prefix is not `/ (index)` but `/dist/esm` which doesn't match any types path.

I found that the simplest approach would be to create a copy of the types in `dist/esm` to have the default TS resolution work for ESM modules.

Reply:
We can be explicit about it, but it is not required since it is automatically resolved based on the file name.